eegstats/R/main.R

```diff
@@ -1,69 +1,299 @@
+      plot(grand_average_erp_data[i,], type = "l", col = colours[i], ylab = 'Amplitude', xlab = 'Time (ms)', xaxt = "n", main = paste('ERP Plot for ', length(cond), 'conditions'))
     }
-    if (i >= 2) {
-      lines(colMeans(GA[,i,]), col = colours[i])
+
+    else if (i >= 2) {
+      lines(grand_average_erp_data[i,], col = colours[i])
+
     }
   }
   legend('topleft', legend=cond, col=colours[1:i], lty=1, cex=0.8)
 
   # add correct timepoints (as in original data)
-  timepoints <- my_erpdata[my_erpdata[,1] == 'S1' & my_erpdata[,2] == 'Negative',3]
-  desired_timepoints <- seq(from = timepoints[1], to = timepoints[length(timepoints)], by = 400)
+  desired_timepoints <- seq(from = time[1], to = time[length(time)], by = 400)
+
+  index_time <- c()
 
-  index_timepoints <- c()
   for (i in 1:length(desired_timepoints)) {
+
     j <- desired_timepoints[i]
-    index_timepoints[i] <- which(timepoints == j)
+
+    index_time[i] <- which(time == j)
+
   }
-  axis(1, at = index_timepoints, labels = desired_timepoints)
+  axis(1, at = index_time, labels = desired_timepoints)
 }
+grand_average_erp_plot(my_erpdata, grand_average_erp_data)
 
 
 
-#' LoadingDate
-#'
-#' @description  The function \emph{GA_Plot} computes the grand average ERP signal for all conditions and plots it.
-#'
-#' @export LoadingData
-#'
-LoadingDate <- function(my_erpdata) {
-  print('hi')
+
+## ERSP data ----
+# ERSP stands for event - related spectral dynamics. Instead of looking at the signal amplitude, here we
+# focus on different frequency bands (e.g. alpha: 8 - 12 Hz). Usually the input for these type of
+# analyses should have the form: subject ID x condition x time x electrodes x frequency.
+# That's a typical shape of pre - processed EEG data.
+
+#To make things a bit easier, we assume that the different electrodes in this dataset (ERPdata)
+# represent frequencies at one electrode. So here we assume that the 20 electrodesrepresent 20
+# different frequencies that have been sampled at one electrode. So electrode 1 would be 5 Hz,
+# electrode 2 would be 6 Hz until electrode 20 represents 25 Hz.
+
+# 5. grand average ersp
+#' grand_average_ersp
+#' @description  The function \emph{grand_average_ersp} computes the grand average  ERSP signal for all conditions and all frequencies from one electrode.
+#' @param my_erspdata Input Matrix with shape: subject ID x condition x timex frequency (from only one electrode).
+#' @details For more details, see Wikipedie page: \href{https://sccn.ucsd.edu/wiki/Chapter_11:_Time/Frequency_decomposition}{article}
+#' @return A matrix that contains the grand average ERSP for all conditions and all frequencies of one electrode. The matrix will be of shape condition x time x frequency.
+#' @export grand_average_ersp_data
+#' @seealso Add example of similar functions [still to be done].
+
+grand_average_ersp <- function(my_erspdata) {
```

Comment:
according to the style guide, function names should be verbs, so maybe change it to something like compute_grand_average_ersp. Even though that is very long but maybe you can come up with something better ;)